components/browser/state/src/main/java/mozilla/components/browser/state/reducer/BrowserStateReducer.kt

```diff
@@ -59,6 +60,7 @@ internal object BrowserStateReducer {
             is CrashAction -> CrashReducer.reduce(state, action)
             is LastAccessAction -> LastAccessReducer.reduce(state, action)
             is UndoAction -> UndoReducer.reduce(state, action)
+            is ShareInternetResourceAction -> ShareInternetResourceStateReducer.reduce(state, action)
```

Comment:
Could we add a test for this line?  See https://github.com/mozilla-mobile/android-components/pull/9420/checks?check_run_id=1747490825

Reply:
Looked into this but I see no reasonable way to test that we are making this call to a static method.
With `mockk` we would've been able to use `mockkStatic` but with `mockito-kotlin` I don't know how to verify this call.
(Would be curious about why AC doesn't also uses `mockk` which is more powerful and whether a change to transition to it is considered.)
Testing the effect - that the state BrowserStateReducer contains what we wanted I think is not a good test since this reducer delegates the behavior (this should be tested) and has no control over the return.

Reply:
That's reasonable, we could open an issue to discuss about mockk on AC  👍🏽

Reply:
Tried using a Robolectric Shadow of `ShareInternetResourceStateReducer` but I get `java.lang.BootstrapMethodError: call site initialization exception`.
Maybe because it is an object? Never used Shadows for objects.

Reply:
As you mentioned above it doesn't worth the while testing it. Yes, to be able to verify we would need to replace the reference with one that we spy.